include/swift/AST/AutoDiff.h

```diff
@@ -363,10 +377,44 @@ class VectorSpace {
 
 namespace llvm {
 
+using swift::AutoDiffConfig;
+using swift::AutoDiffDerivativeFunctionKind;
+using swift::GenericSignature;
+using swift::IndexSubset;
 using swift::SILAutoDiffIndices;
 
 template<typename T> struct DenseMapInfo;
 
+template<> struct DenseMapInfo<AutoDiffConfig> {
+  static AutoDiffConfig getEmptyKey() {
+    auto *ptr = llvm::DenseMapInfo<void *>::getEmptyKey();
+    return {static_cast<IndexSubset *>(ptr),
+            static_cast<IndexSubset *>(ptr),
+            static_cast<GenericSignature *>(ptr)};
+  }
+
+  static AutoDiffConfig getTombstoneKey() {
+    auto *ptr = llvm::DenseMapInfo<void *>::getTombstoneKey();
+    return {static_cast<IndexSubset *>(ptr),
+        static_cast<IndexSubset *>(ptr),
```

Comment:
Fix indentation.

Reply:
Done in https://github.com/apple/swift/pull/27487/commits/771143ae652d5b9cc49fc152863b4d133ee9c9c2.